samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -78,6 +77,14 @@ public StartpointManager(MetadataStoreFactory metadataStoreFactory, Config confi
     LOG.info("StartpointManager created with metadata store: {}", metadataStore.getClass().getCanonicalName());
   }
 
+  /**
+   * Constructs a {@link StartpointManager} instance.
+   */
+  public StartpointManager(CoordinatorStreamStore metadataStore) {
```

Comment:
This class shouldn't be initing/stopping the metadatastore. Same for LocalityManger et. al.